sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaRulesDefinition.java

```diff
@@ -0,0 +1,86 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.sonar;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import org.sonar.api.SonarRuntime;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonarsource.analyzer.commons.RuleMetadataLoader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.sonar.Constants.JSON_PROFILE_PATH;
+import static io.ballerina.sonar.Constants.LANGUAGE_KEY;
+import static io.ballerina.sonar.Constants.RULE_KEY;
+import static io.ballerina.sonar.Constants.RULE_REPOSITORY_KEY;
+import static io.ballerina.sonar.Constants.RULE_REPOSITORY_NAME;
+import static io.ballerina.sonar.Constants.RULE_RESOURCE_FOLDER;
+
+/**
+ * Represents the implementation of the {@link RulesDefinition} class for Ballerina.
+ *
+ * @since 0.1.0
+ */
+public class BallerinaRulesDefinition implements RulesDefinition {
+    private final SonarRuntime runtime;
+    private final Gson gson = new Gson();
+
+    public BallerinaRulesDefinition(SonarRuntime runtime) {
+        this.runtime = runtime;
+    }
+
+    @Override
+    public void define(Context context) {
+        NewRepository repository = context.createRepository(RULE_REPOSITORY_KEY, LANGUAGE_KEY);
+        repository.setName(RULE_REPOSITORY_NAME);
+        RuleMetadataLoader ruleMetadataLoader = new RuleMetadataLoader(RULE_RESOURCE_FOLDER, JSON_PROFILE_PATH,
+                runtime);
+        ruleMetadataLoader.addRulesByRuleKey(repository, loadRuleKeys());
+        repository.done();
+    }
+
+    private List<String> loadRuleKeys() {
+        InputStream inputStream = BallerinaRulesDefinition.class.getResourceAsStream("/ruleKeys.json");
+        StringBuilder builder = new StringBuilder();
+        if (inputStream != null) {
+            try (
+                    InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+                    BufferedReader br = new BufferedReader(inputStreamReader)
+            ) {
+                String content = br.readLine();
+                builder.append(content);
+                while ((content = br.readLine()) != null) {
+                    builder.append(System.lineSeparator()).append(content);
+                }
+            } catch (IOException ex) {
+                throw new IllegalStateException(ex);
+            }
+        }
+        JsonArray rules = gson.fromJson(builder.toString(), JsonArray.class);
+        List<String> ruleKeys = new ArrayList<>();
+        rules.forEach(rule -> ruleKeys.add(rule.getAsJsonObject().get(RULE_KEY).getAsString()));
```

Comment:
Can't we just map the rules instead of explicitly allocating an `ArrayList` here (otherwise let's initialize with correct size)

```suggestion
        List<String> ruleKeys = rules.stream().map(rule -> ruleKeys.add(rule.getAsJsonObject().get(RULE_KEY).getAsString())).toList();
```

Reply:
There are no direct mapping options provided by the `JsonArray` object, so will allocate the ArrayList.